File: app/src/main/java/org/hackillinois/android/utils/FavoritesManager.java
package org.hackillinois.android.utils;

import android.content.Context;
import android.content.SharedPreferences;

import org.hackillinois.android.R;
import org.hackillinois.android.model.Event;

public class FavoritesManager {

    public static void favoriteEvent(Context context, Event event) {
        setBoolean(context, event.getName(), true);
    }

    public static void unfavoriteEvent(Context context, Event event) {
        setBoolean(context, event.getName(), false);
    }

    public static boolean isFavorited(Context context, Event event) {
        return getBoolean(context, event.getName(), false);
    }

    private static void setBoolean(Context context, String key, boolean value) {
        SharedPreferences.Editor editor = getFavoritesPrefs(context).edit();
        editor.putBoolean(key, value);
        editor.apply();
    }

    private static boolean getBoolean(Context context, String key, boolean defValue) {
        return getFavoritesPrefs(context).getBoolean(key, defValue);
    }

    private static SharedPreferences getFavoritesPrefs(Context context) {
        return context.getSharedPreferences(context.getString(R.string.favorites_pref_file_key), Context.MODE_PRIVATE);
    }
}

File: app/src/main/java/org/hackillinois/android/common/FavoritesManager.kt
package org.hackillinois.android.common

import android.content.Context
import org.hackillinois.android.R
import org.hackillinois.android.database.entity.Event
import org.hackillinois.android.notifications.HackIllinoisNotificationManager

class FavoritesManager {
    companion object {
        fun favoriteEvent(context: Context, event: Event?) = event?.let {
            setBoolean(context, it.name, true)
            HackIllinoisNotificationManager.scheduleEventNotification(context, it)
        }

        fun unfavoriteEvent(context: Context, event: Event?) = event?.let {
            setBoolean(context, it.name, false)
            HackIllinoisNotificationManager.cancelEventNotification(context, it)
        }

        fun isFavorited(context: Context, eventName: String) = getBoolean(context, eventName)

        private fun setBoolean(context: Context, key: String, value: Boolean) {
            getFavoritesPrefs(context).edit().apply {
                putBoolean(key, value)
                apply()
            }
        }

        private fun getBoolean(context: Context, key: String) = getFavoritesPrefs(context).getBoolean(key, false)
        private fun getFavoritesPrefs(context: Context) = context.getSharedPreferences(context.getString(R.string.favorites_pref_file_key), Context.MODE_PRIVATE)
    }
}